### DISCUSSION TITLE
How to generate the message required for offline signing

### DISCUSSION THREAD

**Discussion by dqli164** [Chia-Network/chia-blockchain#18565] [Support]
`{
    "signing_instructions": {
      "key_hints": {
        "path_hints": [
          {
            "path": [
              12381,
              8444,
              2,
              7
            ],
            "root_fingerprint": "0xcb86d936"
          }
        ],
        "sum_hints": [
          {
            "final_pubkey": "0x9302b2ace8563b8318699996dfc8d5470b3527e5ead70933bd21d2cdc5821d1eb859ae13366702ba2a2446599a10d88c",
            "fingerprints": [
              "0x0352edbd"
            ],
            "synthetic_offset": "0x2eb8332382fb6b4971e96e3651534ce06b5facb1c326d98863355b1a4473e128"
          }
        ]
      },
      "targets": [
        {
          "fingerprint": "0x4aef2c07",
          "hook": "0xae210bbdb4bf86c4ea49a3eb335310b30b14ca21c82a6aee05cb8fd08c609953",
          "message": "0xfcc1780753ea084e304438752b6c4c6d4d2de855ba7d175730036d36b7664968833eca14ce90215724e61ea40d6706a34c8b7d2b9d3ec9290682b1e95805d4fcccd5bb71183532bff220ba46c268991a3ff07eb358e8255a65c30a2dce0e5fbb"
        }
      ]
    },
    "transaction_info": {
      "spends": [
        {
          "coin": {
            "amount": 249999999832,
            "parent_coin_id": "0x55482d4930b313ced134813ef81162ae9e6360ff5fcb95e214e0433507eeece7",
            "puzzle_hash": "0x2c0327f219b250e0104972478917910767ef134c55b35008658f25bed00133dc"
          },
          "puzzle": "0xff02ffff01ff02ffff01ff02ffff03ff0bffff01ff02ffff03ffff09ff05ffff1dff0bffff1effff0bff0bffff02ff06ffff04ff02ffff04ff17ff8080808080808080ffff01ff02ff17ff2f80ffff01ff088080ff0180ffff01ff04ffff04ff04ffff04ff05ffff04ffff02ff06ffff04ff02ffff04ff17ff80808080ff80808080ffff02ff17ff2f808080ff0180ffff04ffff01ff32ff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff06ffff04ff02ffff04ff09ff80808080ffff02ff06ffff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080ffff04ffff01b09302b2ace8563b8318699996dfc8d5470b3527e5ead70933bd21d2cdc5821d1eb859ae13366702ba2a2446599a10d88cff018080",
          "solution": "0xff80ffff01ffff3cffa0d8bf9fa76f1de74426da5350afde6f7c069103d70a7088a8a810f0f119b34d6f80ffff33ffa0306d77879f997c7902fc095954adf819091efcf84d9b13e0137278c439d943e5ff853a3529432180ffff33ffa08acc5d7f1383b5ea598d2f662c716ae8749730b736b58b24d0c8bae1497bc165ff3280ffff34ff058080ff8080"
        }
      ]
    }
  }`
The above is a regular signing instruction, the message to be signed consists of 

fcc1780753ea084e304438752b6c4c6d4d2de855ba7d175730036d36b7664968

833eca14ce90215724e61ea40d6706a34c8b7d2b9d3ec9290682b1e95805d4fc(the coin id) 

ccd5bb71183532bff220ba46c268991a3ff07eb358e8255a65c30a2dce0e5fbb(the genesis id)

I don't know how fcc1780753ea084e304438752b6c4c6d4d2de855ba7d175730036d36b7664968 is generated, and I can't find any relevant information. it's a random string ?
I am trying to implement offline signing functionality in golang, but I am stuck here. Any help would be appreciated！

**Comment by BrandtH22** [comment]
Hey @dqli164 , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

For reference here is an archived offline signing demo:
https://github.com/Chia-Network/offline-signing-demo

It would also be worthwhile to review the CHIP28 https://github.com/Chia-Network/chips/pull/103/files?short_path=f80be0c#diff-f80be0ceedd49f4f8ff1ae26bed473c90ce50d0e0350296a9d3ddd564bc4baf6 and the other recent signer chips (27-31) as those contain more information about the new signing protocol and how to interact with it.

**Comment by BrandtH22** [comment]
No response from OP, please reopen the ticket if this issue persists
